bgpd/bgp_fsm.c

```diff
@@ -656,6 +689,17 @@ bgp_connect_success (struct peer *peer)
 static int
 bgp_connect_fail (struct peer *peer)
 {
+  if (peer_dynamic_neighbor(peer))
+   {
+      if (BGP_DEBUG (events, EVENTS))
+      {
+        zlog_debug ("%s (dynamic neighbor) deleted", peer->host);
+      }
+
+      peer_delete (peer);
+      return -1;
+   }
+
```

Comment:
The same code 4 times. Probably it's better to introduce a static function here?